apps/users/src/app/app.controller.ts

```diff
@@ -881,4 +881,46 @@ export class UserController {
       return account;
     }
   }
+
+  /**
+   * Update setting to account
+   * @param {CredentialRequest} req Request that has credential from interceptor or passport
+   * @param {UserSettingsDto} body setting dto payload
+   * @returns {''}
+   */
+  @UsePipes(new ValidationPipe({ skipMissingProperties: true }))
+  @ApiResponse({
+    status: HttpStatus.NO_CONTENT
+  })
+  @ApiBody({
+    type: UserSettingsDto
+  })
+  @CastcleBasicAuth()
+  @Put('settings')
+  @HttpCode(HttpStatus.NO_CONTENT)
+  async updateUserSettings(
+    @Req() req: CredentialRequest,
+    @Body() body: UserSettingsDto
+  ) {
+    logger.log(`Start user setting.`);
+    logger.log(JSON.stringify(body));
+    const account = await this.authService.getAccountFromCredential(
+      req.$credential
+    );
+    if (
+      !body ||
+      !body.preferredLanguages ||
+      body.preferredLanguages.length === 0
+    ) {
```

Comment:
```suggestion
    if (!body?.preferredLanguages?.length) {
```

For better readability and shorter condition

ref. [Optional Chaining](https://www.typescriptlang.org/docs/handbook/release-notes/typescript-3-7.html)